Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1639,8 +1639,19 @@ internal IEnumerable<BlockHash> IterateBlockHashes(int offset = 0, int? limit =
         /// <see cref="BlockCommit.Height"/> less than <paramref name="limit"/>.
         /// </summary>
         /// <param name="limit">A exceptional index that is not to be removed.</param>
+        /// <exception cref="InvalidOperationException">Thrown when <see cref="IsCanonical"/> is
+        /// <see langword="false"/>.</exception>
         internal void CleanupBlockCommitStore(long limit)
```

Comment:
I think `DeleteBlockCommit()` became a dead code. Behavior has been changed to putting/getting a `BlockCommit` in/from `_chainDB`, but putting/getting in/from `BlockCommit` is not called, and so `DeleteBlockCommit()` will remove nothing for all time.

Reply:
`DeleteBlockCommit()` is intended to remove a `BlockCommit` from `_blockCommitDB`, which is chain agnostic, not `_chainDB`, which is tied to a specific `Guid`. `BlockCommit`s are cached during download during preloading phase.

Reply:
Ah, yes, I forgot that a `BlockCommit` is stored while in synchronization.